crates/sui-bridge/src/orchestrator.rs

```diff
@@ -73,39 +94,59 @@ where
                 .collect::<BridgeResult<Vec<_>>>()
                 .expect("Sui Event could not be deserialzed to SuiBridgeEvent");
 
-            // Load committee upfront to avoid weird edge cases where committee changes in between
-            let _committee = bridge_committee.load().clone();
-
-            // TODO: optimize handling of multiple events
+            let mut actions = vec![];
             for (sui_event, opt_bridge_event) in events.iter().zip(bridge_events) {
                 if opt_bridge_event.is_none() {
                     // TODO: we probably should not miss any events, warn for now.
                     warn!("Sui event not recognized: {:?}", sui_event);
                 }
-                let _bridge_event = opt_bridge_event.unwrap();
+                let bridge_event: SuiBridgeEvent = opt_bridge_event.unwrap();
 
-                // TODO: handle all bridge events
+                if let Some(action) = bridge_event
+                    .try_into_bridge_action(sui_event.id.tx_digest, sui_event.id.event_seq as u16)
+                {
+                    actions.push(action);
+                }
+                // TODO: handle non Action events
             }
+
+            if !actions.is_empty() {
+                // Write action to pending WAL
+                store
+                    .insert_pending_actions(&actions)
```

Comment:
Would we be executing the action twice if server restarts after executing an action. Maybe the actions are idempotent and hence safe, but do we need exactly once action execution?

Reply:
the actions are indeed idempotent. I think we don't need exactly once guarantee here (it's going to be guaranteed by Move code)